### PR TITLE
feat: new auth flow + new build process

### LINKED ISSUE #85: Support multiple login options
- Support for the same login options the rest of the product supports
- Support to connect with a Gnosis multisig through WalletConnect

### PR DESCRIPTION
**Pull request by andyesp**
Closes #85

### CODE REVIEW

File: src/components/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -11,41 +11,45 @@ import useResponsive, { onlyMobileMaxWidth } from '../../hooks/useResponsive'
...
           {isMobile ? (
             <Grid>
               <Grid.Row columns={1}>
                 <Grid.Column>
-                  <Schedule />
+                  <Schedule contract={contract} />
                 </Grid.Column>
                 <Grid.Column className="detailsMobile">
-                  <Details />
+                  <Details address={'0xuseraddress'} contract={contract} />
```
Comment (ncomerci) [inline src/components/Dashboard/Dashboard.tsx:45]: `0xuseraddress`

File: src/components/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -11,41 +11,45 @@ import useResponsive, { onlyMobileMaxWidth } from '../../hooks/useResponsive'
...
                 </Grid.Column>
               </Grid.Row>
             </Grid>
           ) : (
             <>
-              <Schedule />
-              <Details />
+              <Schedule contract={contract} />
+              <Details address={'0xuseraddress'} contract={contract} />
```
Comment (ncomerci) [inline src/components/Dashboard/Dashboard.tsx:52]: `0xuseraddress`

File: src/components/Dashboard/Details/Details.tsx [hunk 3]
```diff
@@ -33,10 +35,12 @@ function getBeneficiary(addr) {
 }
 
 function getDate(id, date) {
+  // TODO: Check if date is correct
```
Comment (ncomerci) [inline src/components/Dashboard/Details/Details.tsx:38]: TODO

File: src/config/env/dev.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "NETWORK": "sepolia",
```
Comment (ncomerci) [inline src/config/env/dev.json:2]: do we have vesting contracts in sepolia?

File: src/config/env/prod.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "NETWORK": "mainnet",
+  "CHAIN_ID": "1",
+  "SEGMENT_API_KEY": "TODO",
```
Comment (ncomerci) [inline src/config/env/prod.json:4]: TODO

File: src/modules/identity/sagas.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  yield takeEvery(loginRequest.type, handleLogin)
+  yield takeEvery(CHANGE_ACCOUNT, handleChangeAccount)
+  yield takeEvery(CONNECT_WALLET_SUCCESS, handleConnectWalletSuccess)
+  yield takeEvery(DISCONNECT_WALLET, handleDisconnectWallet)
+
+  function* handleLogin(action: LoginRequestAction) {
+    const providerType = action.payload
+    // Check if we need to generate an identity
+    try {
+      // Check if we need to connect the wallet
```
Comment (ncomerci) [inline src/modules/identity/sagas.ts:35]: TODO

File: src/utils/index.js [hunk 7]
```diff
@@ -10,10 +10,17 @@ export function isValidAddress(address) {
...
 export function getPreciseDiff(start, finish) {
-  const diff = moment.preciseDiff(moment(start * 1000), moment(finish * 1000), true)
+  // const diff = moment.preciseDiff(moment(start * 1000), moment(finish * 1000), true)
+  // return {
+  //   months: diff.years * 12 + diff.months,
+  //   days: diff.days,
+  // }
+
+  // TODO: Fix This
+
```
Comment (ncomerci) [inline src/utils/index.js:20]: TODO

File: vite.config.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
+// import { NodeGlobalsPolyfillPlugin } from '@esbuild-plugins/node-globals-polyfill'
+// import { NodeModulesPolyfillPlugin } from '@esbuild-plugins/node-modules-polyfill'
+import react from '@vitejs/plugin-react-swc'
+// import rollupNodePolyFill from 'rollup-plugin-polyfill-node'
```
Comment (ncomerci) [inline vite.config.ts:4]: 🧹

**Review by ncomerci — APPROVED** [review]

File: .github/workflows/set-rollout.yaml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+          packageVersion: ${{ github.event.deployment.payload.packageVersion }}
+
+          # Rollout information
+          deploymentPath: 'vesting-dashboard'
+          deploymentEnvironment: 'zone'
+          deploymentName: '_site'
+          percentage: 100
+
+      # Stg
+      - name: Set Rollout - Staging
```
Comment (LautaroPetaccio) [inline .github/workflows/set-rollout.yaml:33]: This job deploys the site to staging, but you're not enabling the staging environment in the definitions repo, would you mind removing it?

File: .github/workflows/build-release.yaml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+          cwd: './dist'
+          deterministic-snapshot: true
+          provenance: false
+          registry-url: 'https://registry.npmjs.org'
+          access: public
+          gitlab-token: ${{ secrets.GITLAB_CDN_DEPLOYER_TOKEN }}
+          gitlab-pipeline-url: ${{ secrets.GITLAB_CDN_DEPLOYER_URL }}
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - name: Create Sentry release
```
Comment (LautaroPetaccio) [inline .github/workflows/build-release.yaml:46]: Has the site sentry configured?

File: .github/workflows/build-release.yaml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Publish
+        uses: menduz/oddish-action@master
+        with:
+          cwd: './dist'
+          deterministic-snapshot: true
+          provenance: false
+          registry-url: 'https://registry.npmjs.org'
+          access: public
+          gitlab-token: ${{ secrets.GITLAB_CDN_DEPLOYER_TOKEN }}
+          gitlab-pipeline-url: ${{ secrets.GITLAB_CDN_DEPLOYER_URL }}
```
Comment (LautaroPetaccio) [inline .github/workflows/build-release.yaml:43]: Would you mind asking for a devops admin to add these secrets to the repository?

File: .github/workflows/build-release.yaml [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+        run: npm run build
+        env:
+          NODE_PATH: 'src'
+          NODE_OPTIONS: '--max-old-space-size=6144'
+      - name: Publish
+        uses: menduz/oddish-action@master
+        with:
+          cwd: './dist'
+          deterministic-snapshot: true
+          provenance: false
```
Comment (LautaroPetaccio) [inline .github/workflows/build-release.yaml:39]: You can remove the provenance variable here if you set the correct permissions in the job.
```suggestion
```

File: .github/workflows/build-release.yaml [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+on:
+  push:
+    branches:
+      - master
+  release:
+    types:
+      - created
+
+jobs:
+  build-release:
```
Comment (LautaroPetaccio) [inline .github/workflows/build-release.yaml:12]: This permission should be here in order to enable the provenance feature:
```suggestion
  build-release:
      permissions:
      id-token: write
```

File: .github/workflows/build-release.yaml (hunk 11 above)
Comment (andyesp) [reply .github/workflows/build-release.yaml:44]: Done!

Comment (andyesp) [reply .github/workflows/build-release.yaml:44]: Secrets already added!

**Pull request merged**